### ISSUE TITLE
Change API to builder pattern

### ISSUE THREAD

**Issue by roniemartinez** [spider-rs/spider#115]
It looks like as we add more features to `Website`, it gets more complicated. Would it be an option to change it to builder pattern, instead, so it would be much simpler to use?

```rust
let mut website: Website = Website::new("https://choosealicense.com");

website.configuration.respect_robots_txt = true;
website.configuration.subdomains = true;
website.configuration.tld = false;
website.configuration.delay = 0; // Defaults to 0 ms due to concurrency handling
website.configuration.request_timeout = None; // Defaults to 15000 ms
website.configuration.http2_prior_knowledge = false; // Enable if you know the webserver supports http2
website.configuration.channel_buffer = 100; // Defaults to 50 - tune this depending on on_link_find_callback
website.configuration.user_agent = Some("myapp/version".into()); // Defaults to using a random agent
website.on_link_find_callback = Some(|s| { println!("link target: {}", s); s }); // Callback to run on each link find
website.configuration.blacklist_url.get_or_insert(Default::default()).push("https://choosealicense.com/licenses/".into());
website.configuration.proxies.get_or_insert(Default::default()).push("socks5://10.1.1.1:12345".into()); // Defaults to none - proxy list.

website.crawl().await;
```

When setting up with builder pattern (suggestion only):

```rust
let mut website: Website = Website::new("https://choosealicense.com")
    .with_respect_robots_txt(true)
    .with_subdomains(true)
    .with_tld(false)
    .with_delay(0)
    .with_request_timeout(None)
    .with_http2_prior_knowledge(false)
    .with_channel_buffer(100)
    .with_user_agent(Some("myapp/version".into()))
    .with_on_link_find_callback( Some(|s| { println!("link target: {}", s); s }))
    .with_blacklist_url("https://choosealicense.com/licenses/".into())
    .with_blacklist_url( .... ) // can add more
    .with_proxy("socks5://10.1.1.1:12345".into())
    .with_proxy( ... ); // can add more

website.crawl().await;
```

**Comment by j-mendez** [comment]
@roniemartinez you are on the money with this one. I think this is a good idea. The glob feature looked simple at first, except when it came to the subtle differences needed per feature. It went from a 1/5 level of effort to 3/5. I’m glad we had test setup and made sure to add more tests to catch all the edges.

**Comment by j-mendez** [comment]
@roniemartinez Want to do a PR for a module that updates the configuration like SpiderBuilder? I can also add this too later. I think it’s a good addition. The SpiderBuilder can use the names of the configuration as is for clarity. Some of the docs need updating on the feature that makes sense for the configurations too, so this brings a good place for a clean entry.

**Comment by roniemartinez** [comment]
@j-mendez I only have time again on next weekend.

**Comment by j-mendez** [comment]
@roniemartinez thank you for the issue this is now available on [v1.33.0](https://github.com/spider-rs/spider/releases).
